bindings/rust/bench/src/rustls.rs

```diff
@@ -35,42 +41,43 @@ impl RustlsConnection {
         &self.connection
     }
 
-        /// Treat `WouldBlock` as an `Ok` value for when blocking is expected
-        fn ignore_block<T: Default>(res: Result<T, std::io::Error>) -> Result<T, std::io::Error> {
-            match res {
-                Ok(t) => Ok(t),
-                Err(err) => match err.kind() {
-                    std::io::ErrorKind::WouldBlock => Ok(T::default()),
-                    _ => Err(err),
-                },
-            }
+    /// Treat `WouldBlock` as an `Ok` value for when blocking is expected
+    fn ignore_block<T: Default>(res: Result<T, std::io::Error>) -> Result<T, std::io::Error> {
+        match res {
+            Ok(t) => Ok(t),
+            Err(err) => match err.kind() {
+                std::io::ErrorKind::WouldBlock => Ok(T::default()),
+                _ => Err(err),
+            },
         }
+    }
 }
 
 impl RustlsConfig {
-    fn get_root_cert_store(sig_type: SigType) -> Result<RootCertStore, Box<dyn Error>> {
-        let root_cert =
-            Certificate(certs(&mut BufReader::new(&*read_to_bytes(CACert, sig_type)))?.remove(0));
-        let mut root_certs = RootCertStore::empty();
-        root_certs.add(&root_cert)?;
-        Ok(root_certs)
-    }
-
-    fn get_cert_chain(
-        pem_type: PemType,
-        sig_type: SigType,
-    ) -> Result<Vec<Certificate>, Box<dyn Error>> {
-        let chain = certs(&mut BufReader::new(&*read_to_bytes(pem_type, sig_type)))?;
-        Ok(chain
-            .iter()
-            .map(|bytes| Certificate(bytes.to_vec()))
-            .collect())
-    }
-
-    fn get_key(pem_type: PemType, sig_type: SigType) -> Result<PrivateKey, Box<dyn Error>> {
-        Ok(PrivateKey(
-            pkcs8_private_keys(&mut BufReader::new(&*read_to_bytes(pem_type, sig_type)))?.remove(0),
-        ))
+    fn get_root_cert_store(sig_type: SigType) -> RootCertStore {
+        let mut root_store = RootCertStore::empty();
+        root_store.add_parsable_certificates(
+            rustls_pemfile::certs(&mut BufReader::new(&*read_to_bytes(CACert, sig_type)))
+                .map(|r| r.unwrap()),
+        );
+        root_store
+    }
+
+    fn get_cert_chain(pem_type: PemType, sig_type: SigType) -> Vec<CertificateDer<'static>> {
+        rustls_pemfile::certs(&mut BufReader::new(&*read_to_bytes(pem_type, sig_type)))
+            .map(|result| result.unwrap())
+            .collect()
+    }
+
+    fn get_key(pem_type: PemType, sig_type: SigType) -> PrivateKeyDer<'static> {
+        if let Some(rustls_pemfile::Item::Pkcs8Key(key)) =
+            rustls_pemfile::read_one(&mut BufReader::new(&*read_to_bytes(pem_type, sig_type)))
+                .unwrap()
+        {
+            return key.into();
+        } else {
+            panic!("unsupported key type");
```

Comment:
A failure would mean supporting additional formats. Can you mention/link to the [docs](https://docs.rs/rustls-pemfile/latest/rustls_pemfile/enum.Item.html)